### PR TITLE
doc: add kernel/thread stack usage notes

### PR DESCRIPTION
**Pull request by tyrantlizard77**
This is covered in the porting guide, and we have complete docstrings
for these macros, but having discussion here helps new users understand
how to use these and why they exist.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

### CODE REVIEW

**Review by nashif — APPROVED** [review]

File: doc/reference/kernel/threads/index.rst [hunk 1]
```diff
@@ -301,9 +357,22 @@ Spawning a Thread
...
+
+- The original requested stack size passed to
+  ``K_THREAD_STACK`` or ``K_KERNEL_STACK`` family of stack instantiation
+  macros.
+- For a stack object defined with the ``K_THREAD_STACK`` family of
+  macros, the return value of :c:macro:`K_THREAD_STACK_SIZEOF()` for that'
+  object.
+- For a stack object defined with the ``K_KERNEL_STACK`` family of
+  macros, the return value of :c:macro:`K_KERNEL_STACK_SIZEOF()` for that
+  object.
```
Comment (ceolin) [inline doc/reference/kernel/threads/index.rst:375]: Second and third are the same

Comment (ceolin) [reply doc/reference/kernel/threads/index.rst:375]: The doxygen API documentation just says to use `K_KERNEL_STACK_SIZEOF`, maybe it is worth to add the other possibilities there as well.

**Review by ceolin — CHANGES_REQUESTED** [review]
There is duplicated contents.

**Review by ruuddw — APPROVED** [review]

File: doc/reference/kernel/threads/index.rst (hunk 1 above)
Comment (andrewboie) [reply doc/reference/kernel/threads/index.rst:375]: @ceolin no they aren't the same.

One discusses `K_THREAD_STACK_*` macros, the other `K_KERNEL_STACK`

Comment (ceolin) [reply doc/reference/kernel/threads/index.rst:375]: What shame, just read the same thing. Sorry.

**Review by ceolin — APPROVED** [review]

**Pull request merged**